x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -427,42 +432,42 @@ public static String createRollupIndexMapping(
+                assert indexScopedSettings.isPrivateSetting(key) : "expected [" + key + "] to be private but it was not";
+            } else if (setting.getProperties().contains(Setting.Property.NotCopyableOnResize)) {
+                // we leverage the NotCopyableOnResize setting property for rollup, because
+                // the same rules with resize apply
+                continue;
+            }
+            // do not override settings that have already been set in the rollup index
+            if (targetSettings.keys().contains(key)) {
+                continue;
+            }
+            targetSettings.copy(key, sourceIndexMetadata.getSettings());
+        }
 
         /*
          * Add the source index name and UUID to the rollup index metadata.
          * If the source index is a rollup index, we will add the name and UUID
          * of the first index that we initially rolled up.
          */
-        String originalIndexName = IndexMetadata.INDEX_ROLLUP_SOURCE_NAME.exists(sourceIndexMetadata.getSettings())
-            ? IndexMetadata.INDEX_ROLLUP_SOURCE_NAME.get(sourceIndexMetadata.getSettings())
-            : sourceIndexName;
-        String originalIndexUuid = IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.exists(sourceIndexMetadata.getSettings())
-            ? IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.get(sourceIndexMetadata.getSettings())
-            : sourceIndexMetadata.getIndexUUID();
-
-        // Copy time series index settings from original index
-        List<String> indexRoutingPath = sourceIndexMetadata.getRoutingPaths();
-        Instant startTime = IndexSettings.TIME_SERIES_START_TIME.get(sourceIndexMetadata.getSettings());
-        Instant endTime = IndexSettings.TIME_SERIES_END_TIME.get(sourceIndexMetadata.getSettings());
-        IndexMode indexMode = IndexSettings.MODE.get(sourceIndexMetadata.getSettings());
-
-        return IndexMetadata.builder(rollupIndexMetadata)
-            .settings(
-                Settings.builder()
-                    .put(rollupIndexMetadata.getSettings())
-                    .put(IndexMetadata.INDEX_ROLLUP_SOURCE_NAME.getKey(), originalIndexName)
-                    .put(IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.getKey(), originalIndexUuid)
-                    .put(IndexMetadata.INDEX_HIDDEN_SETTING.getKey(), sourceIndexMetadata.isHidden())
-                    // Add the time series index settings
-                    .put(IndexSettings.MODE.getKey(), indexMode)
-                    .putList(IndexMetadata.INDEX_ROUTING_PATH.getKey(), indexRoutingPath)
-                    .put(IndexSettings.TIME_SERIES_START_TIME.getKey(), startTime.toString())
-                    .put(IndexSettings.TIME_SERIES_END_TIME.getKey(), endTime.toString())
-            );
+        if (IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.exists(sourceIndexMetadata.getSettings()) == false
+            || IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.exists(sourceIndexMetadata.getSettings()) == false) {
```

Comment:
This is the same check twice, is one of these supposed to be checking the `*_NAME` instead of `*_UUID`?

Reply:
Good catch, it was `_NAME`and `_UUID` . Fixing it now.

Reply:
Fixed in [02c856d](https://github.com/elastic/elasticsearch/pull/88565/commits/02c856deba6986f5199c928b1dd829a667e53abc)